pkg/controller/projectreference/projectreference_adapter.go

```diff
@@ -406,6 +465,34 @@ func (r *ReferenceAdapter) deleteProject() error {
 	}
 }
 
+func (r *ReferenceAdapter) deleteServiceAccount() error {
+	serviceAccountName := r.ProjectReference.Spec.ServiceAccountName
+
+	r.logger.V(1).Info("SA delete started", "serviceAccountName", serviceAccountName)
+
+	sa, err := r.gcpClient.GetServiceAccount(serviceAccountName)
+	if err != nil {
+		// TODO: for not this always occurs, not sure why
```

Comment:
What's this TODO about?

Reply:
there was an issue with deleting the project before the serviceaccount that is didn't clear the serviceaccount. I switched them around and it worked IIRC